src/Take.Blip.Builder/Strategies/CrmContext.cs

```diff
@@ -0,0 +1,28 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+using Take.Blip.Builder.Models;
+using Take.Blip.Builder.Utils;
+
+namespace Take.Blip.Builder.Strategies
+{
+    public class CrmContext : ICrmContext
+    {
+        private ICrmProcessor _crmProcessor;
+        public async Task ExecuteAsync(IContext context, CrmSettings settings, CrmActionType actionType, CancellationToken cancellationToken)
+        {
+            if (actionType.Equals(CrmActionType.GetLead))
+            {
+                await _crmProcessor.GetLeadAsync(context, settings, cancellationToken);
+            }
+            else if (actionType.Equals(CrmActionType.CreateLead))
+            {
+                await _crmProcessor.RegisterLeadAsync(context, settings, cancellationToken);
+            }
+        }
+
+        public void SetCrm(ICrmProcessor crmProcessor)
+        {
+            _crmProcessor = crmProcessor;
+        }
```

Comment:
You could use a property:

```csharp
public string Crm { private get; set; }
```

Reply:
I dont think using a property instead of setter method has any gain on readbility or perfomance. The property will not be propagated to interface and if I set property as string more conditionals statement will have to be added. Reduce conditional statements is the main purpose of Strategy Pattern.